setuptools/build_meta.py

```diff
@@ -51,7 +53,9 @@ def __init__(self, specifiers):
 
 class Distribution(setuptools.dist.Distribution):
     def fetch_build_eggs(self, specifiers):
-        raise SetupRequirementsError(specifiers)
+        specifier_list = self._parse_requirements(specifiers)
+
+        raise SetupRequirementsError(specifier_list)
```

Comment:
Why duplicate pkg_resources' code for parsing requirements? Why not use instead:
```python
specifier_list = list(map(str, pkg_resources.parse_requirements(specifiers)))
```
?